mcdc/iqmc/iqmc_kernel.py

```diff
@@ -0,0 +1,714 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Jun 10 21:52:17 2024
+
+@author: sam pasmann
+"""
+from mpi4py import MPI
+
+import mcdc.adapt as adapt
+import numpy as np
+
+from numpy import ascontiguousarray as cga
+from numba import njit, objmode, literal_unroll
+from mcdc.loop import caching
+from mcdc.type_ import iqmc_score_list
+from mcdc.kernel import distance_to_boundary, distance_to_mesh
+
+# from mcdc.iqmc.iqmc_loop import iqmc_loop_source
+from mcdc.adapt import toggle, for_cpu, for_gpu
+from mcdc.constant import *
+from mcdc.kernel import (
+    move_particle,
+    set_bank_size,
+    get_particle_cell,
+    get_particle_material,
+    mesh_get_index,
+)
+
+# =========================================================================
+# Low-Discrepency Sequences
+# =========================================================================
+
+
+@toggle("iQMC")
+def lds_init(mcdc):
+    N, dim = mcdc["technique"]["iqmc"]["lds"].shape
+    N_start = mcdc["mpi_work_start"]
+    mcdc["technique"]["iqmc"]["lds"] = halton(N, dim, skip=N_start)
+
+
+@toggle("iQMC")
+def scramble_LDS(mcdc):
+    # TODO: use MCDC seed system
+    seed_batch = np.random.randint(1000, 1000000)
+    N, dim = mcdc["technique"]["iqmc"]["lds"].shape
+    N_start = mcdc["mpi_work_start"]
+    mcdc["technique"]["iqmc"]["lds"] = rhalton(N, dim, seed=seed_batch, skip=N_start)
+
+
+@toggle("iQMC")
+def rhalton(N, dim, seed=12345, skip=0):
```

Comment:
Could you add some doc strings here? It doesn't need to be as robust as the input deck. The halton sequence seems to be undergriding iQMC so it seems mildly important to me but lmk